Dockerfile

```
@@ -24,19 +59,38 @@ RUN yarn --prod=false --frozen-lockfile \
       --ignore-scripts \
     && yarn cache clean
 
-
 COPY service.js /app/source/
 
-# verify node-rdkafka is installed right
-RUN node -e "require('node-rdkafka')"
 
-# verify teraslice is installed right
-RUN node -e "require('teraslice')"
+FROM node:${NODE_VERSION}-bookworm-slim
+
+# Affects garbage collection. This default gets overwritten by the memory setting in kubernetes
+ENV NODE_OPTIONS "--max-old-space-size=2048"
+ENV NODE_ENV production
 
 EXPOSE 5678
 
 # set up the volumes
 VOLUME /app/config /app/logs /app/assets
 ENV TERAFOUNDATION_CONFIG /app/config/teraslice.yaml
 
+# Use tini to handle sigterm and zombie processes
+ENTRYPOINT ["/usr/bin/tini", "--"]
+
 CMD ["node", "service.js"]
+
+RUN apt-get update && \
+     apt-get install -y libcurl4 tini
```

```suggestion
     apt-get install -y libcurl4 tini && \
       apt-get autoremove -y && \
       apt-get clean -y && \
       rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```

You forget to do the cleanup after the `apt update/install`.  So you're leaving at least 20MB in the image:

```
du -sh /var/lib/apt/lists/ /tmp/ /var/tmp/
19M	/var/lib/apt/lists/
2.2M	/tmp/
4.0K	/var/tmp/
```